src/modules/selectors/select.rs

```diff
@@ -0,0 +1,100 @@
+// Copyright 2013 The Noise-rs Developers.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use num::{Float, Zero};
+use math;
+use Module;
+
+/// Noise module that outputs the value selected from one of two source
+/// modules chosen by the output value from a control module.
+///
+/// - Source Module 1 outputs a value.
+/// - Source Module 2 outputs a value.
+/// - Control Module determines the value to select. If the output value
+/// from the control module is within a range of values known as the selection
+/// range, this noise module outputs the value from Source Module 2. Otherwise,
+/// this noise module outputs the value from Source Module 1.
+pub struct Select<Lhs, Rhs, Control, U> {
+    source1: Lhs,
+    source2: Rhs,
+    selector: Control,
+    edge_falloff: U,
+    lower_bound: U,
+    upper_bound: U,
+}
+
+impl<Lhs, Rhs, Control, U> Select<Lhs, Rhs, Control, U> {
+    pub fn new(s1: Lhs, s2: Rhs, c: Control, falloff: U, lower: U, upper: U) -> Select<Lhs, Rhs, Control, U> {
+        Select {
+            source1: s1,
+            source2: s2,
+            selector: c,
+            edge_falloff: falloff,
+            lower_bound: lower,
+            upper_bound: upper,
+        }
+    }
+}
+
+/// Maps a value onto a cubic S-curve.
+fn s_curve3<T: Float>(a: T) -> T {
+    a * a * (math::cast::<_, T>(3.0) - (math::cast::<_, T>(2.0) * a))
+}
+
+/// Performs linear interploation between two values.
+fn linear_interp<T: Float>(lhs: T, rhs: T, a: T) -> T {
+    ((T::one() - a) * lhs) + (a * rhs)
+}
```

Comment:
I'll move them to a separate interp module, so I can put other interpolation methods in there as well.